src/spikeinterface/extractors/neoextractors/intan.py

```diff
@@ -27,6 +27,8 @@ class IntanRecordingExtractor(NeoBaseRecordingExtractor):
         If True, data that violates integrity assumptions will be loaded. At the moment the only integrity
         check we perform is that timestamps are continuous. Setting this to True will ignore this check and set
         the attribute `discontinuous_timestamps` to True in the underlying neo object.
+    use_names_as_ids : bool or None, default: None
+        If True, use channel names as IDs. If None, use default IDs.
```

Comment:
What are the default IDs? `np.arange(num_chans)`?

Reply:
They are inherited from neo's signal_channels["id"], and seems to depend on the Extractor. See here: https://github.com/NeuralEnsemble/python-neo/blob/1a7f60e2d75176b6df00fd7a2d1344772d223174/neo/rawio/intanrawio.py#L228 for the intan one. Maybe it's enough to just say that it's inherited from neo? And probably put the default first. Something like:

`If False, use default IDs inherited from neo. If True, use channel names as IDs.`

But thanks for pointing this out: just realised the default is incorrect for lots of these. Once we decide on the wording for this, I'll update all of them.

Reply:
Hey guys, neo has 3 things
channel_names: non-immutable (could be set by user) and so we can guarantee uniqueness
channel_ids: the hardware "names" of the recording apparatus sometimes just stringified indexes of the channels if the recording equipment doesn't provide unique names
channel_indices: the indices of the names and ids such that channel_names[1] is the same as channel_ids[1]

Basically setting that to True, the channel_names are used which lose the safety guarantees of neo but are clearer if the user actually "renamed" their channels. For example Intan has a native name which is assigned by the hardware and a custom name which is either the native name or a user specified name.

Reply:
Thanks @zm711! thats super useful / informative. In that case I think @chrishalcrow it would be enough to say default channel IDs from NEO. BTW (out of scope comment for this PR): should `use_names_as_ids` just accept bool only? (i.e. `None` and `False` are presumably redundant).